### PR TITLE
fix: make Jackson annotations work when using hilla:generate

### PR DESCRIPTION
**Pull request by platosha**

### CODE REVIEW

File: packages/java/maven-plugin/src/main/java/dev/hilla/maven/EngineGenerateMojo.java [hunk 1]
```diff
@@ -44,7 +44,8 @@ public void execute() throws EngineGenerateMojoException {
             for (var classPathItem : classPath) {
                 urls.add(new File(classPathItem).toURI().toURL());
             }
```
Comment (Lodin) [inline packages/java/maven-plugin/src/main/java/dev/hilla/maven/EngineGenerateMojo.java:46]: I guess you could use `stream()` here.

File: packages/java/maven-plugin/src/main/java/dev/hilla/maven/EngineConfigureMojo.java [hunk 2]
```diff
@@ -50,11 +50,8 @@ public final class EngineConfigureMojo extends AbstractMojo {
     public void execute() throws EngineConfigureMojoException {
         try {
             EngineConfiguration conf = new EngineConfiguration();
-            conf.setClassPath(Stream
-                    .of(project.getCompileClasspathElements(),
-                            project.getRuntimeClasspathElements())
-                    .flatMap(Collection::stream)
-                    .collect(Collectors.toCollection(LinkedHashSet::new)));
+            conf.setClassPath(
+                    new LinkedHashSet<>(project.getRuntimeClasspathElements()));
```
Comment (Lodin) [inline packages/java/maven-plugin/src/main/java/dev/hilla/maven/EngineConfigureMojo.java:54]: Do we ignore compile-time classpath elements?

Comment (platosha) [reply packages/java/maven-plugin/src/main/java/dev/hilla/maven/EngineConfigureMojo.java:54]: Not really, according to [the source](https://github.com/apache/maven/blob/b82479bf111e4d0dc806d9be6d87f1fec4c2ba40/maven-core/src/main/java/org/apache/maven/project/MavenProject.java#L365), Maven does include compile-time artifacts there.

Comment (Lodin) [reply packages/java/maven-plugin/src/main/java/dev/hilla/maven/EngineConfigureMojo.java:54]: That's... weird 🤨

File: packages/java/maven-plugin/src/main/java/dev/hilla/maven/EngineGenerateMojo.java (hunk 1 above)
Comment (platosha) [reply packages/java/maven-plugin/src/main/java/dev/hilla/maven/EngineGenerateMojo.java:46]: The `toURL()` method throws `MalformedURLException`, and exception handling for Stream items is not nice. Luckily, with good'ol plain code this exception is handled with the surrounding `try / catch`.

Comment (Lodin) [reply packages/java/maven-plugin/src/main/java/dev/hilla/maven/EngineGenerateMojo.java:46]: ah, I see. Meh, that broken Stream...

**Review by Lodin — APPROVED** [review]

**Pull request merged**